core/trino-main/src/main/java/io/trino/connector/system/jdbc/ColumnJdbcTable.java

```diff
@@ -267,7 +273,9 @@ public RecordCursor cursor(ConnectorTransactionHandle transactionHandle, Connect
                     QualifiedTablePrefix tablePrefix = tableFilter.isPresent()
                             ? new QualifiedTablePrefix(catalog, schema, tableFilter.get())
                             : new QualifiedTablePrefix(catalog, schema);
-                    Set<SchemaTableName> tables = listTables(session, metadata, accessControl, tablePrefix);
+                    Set<SchemaTableName> tables = union(
+                            listTables(session, metadata, accessControl, tablePrefix),
+                            listMaterializedViews(session, metadata, accessControl, tablePrefix));
```

Comment:
This PR doesn't change `listTables` implementation yet. This will be subsequent PR

Reply:
Then we at least need to reach agreement for the general direction. see https://github.com/trinodb/trino/pull/8131#issuecomment-851179231

Reply:
I've updated this PR to avoid explicitly calling listMVs in information_schema or system.jdbc
So the code in ColumnJdbc, InformationSchemaMetadata and TableJdbc remains unchanged